File: src/main/java/com/groupdocs/comparison/text/settings/ComparisonWithDifferentSettings.java
package com.groupdocs.comparison.text.settings;

import com.groupdocs.comparison.common.comparisonsettings.ComparisonDepth;
import com.groupdocs.comparison.common.comparisonsettings.TextComparisonSettings;
import com.groupdocs.comparison.text.ComparisonTextFile;
import com.groupdocs.comparison.text.contracts.IComparisonTextFile;
import com.groupdocs.comparison.text.contracts.ITextCompareResult;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public class ComparisonWithDifferentSettings {

    public void CompareTextFilesWithGenerationSummaryPage() throws IOException {
        String resultPath = "./../../Settings/testresult/WithGenerationSummaryPage/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setGenerateSummaryPage(true);
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithOutGenerationSummaryPage() throws IOException {
        String resultPath = "./../../Settings/testresult/WithOutGenerationSummaryPage/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setGenerateSummaryPage(false);
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithShowDeletedContent() throws IOException {
        String resultPath = "./../../Settings/testresult/WithShowDeletedContent/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setShowDeletedContent(true);
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithOutShowDeletedContent() throws IOException {
        String resultPath = "./../../Settings/testresult/WithOutShowDeletedContent/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setShowDeletedContent(false);
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithComparisonDepthSetWords() throws IOException {
        String resultPath = "./../../Settings/testresult/WithComparisonDepthSetWords/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setComparisonDepth(ComparisonDepth.Words);
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithComparisonDepthSetChars() throws IOException {
        String resultPath = "./../../Settings/testresult/WithComparisonDepthSetChars/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setComparisonDepth(ComparisonDepth.Chars);
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithWordsSepCharsSetSpace() throws IOException {
        String resultPath = "./../../Settings/testresult/WithWordsSepCharsSetSpace/result.txt";
        // Creating settings for comparison of textFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.setWordsSeparatorChars(new char[]{' '});
        Compare(resultPath, comparisonSettings);
    }

    public void CompareTextFilesWithSettingStylesOnDelInsComponents() throws IOException {
        String resultPath = "./../../Settings/testresult/WithSettingStylesOnDelInsComponents/result.txt";
        // Creating settings for comparison of TextFiles
        TextComparisonSettings comparisonSettings = new TextComparisonSettings();
        comparisonSettings.getDeletedItemsStyle().setBeginSeparatorString("<begin deleted text>");
        comparisonSettings.getDeletedItemsStyle().setEndSeparatorString("<end deleted text>");
        comparisonSettings.getInsertedItemsStyle().setBeginSeparatorString("<begin inserted text>");
        comparisonSettings.getInsertedItemsStyle().setEndSeparatorString("<end inserted text>");
        Compare(resultPath, comparisonSettings);
    }

    private void Compare(String resultPath, TextComparisonSettings comparisonSettings) throws FileNotFoundException, IOException {
        String sourcePath = "Groupdocs.Comparison.Samples.Text.Settings.data.source.txt";
        String targetPath = "Groupdocs.Comparison.Samples.Text.Settings.data.target.txt";
        // Create to streams of textFiles
        InputStream sourceStream = this.getClass().getClassLoader().getResourceAsStream(sourcePath);
        InputStream targetStream = this.getClass().getClassLoader().getResourceAsStream(targetPath);
        // Opening two textFiles
        IComparisonTextFile sourcePresentation = new ComparisonTextFile(sourceStream);
        System.out.println("TextFile with source path: " + sourcePath + " was loaded.");
        IComparisonTextFile targetPresentation = new ComparisonTextFile(targetStream);
        System.out.println("TextFile with source path: " + targetPath + " was loaded.");

        // Comparing textFiles
        ITextCompareResult compareResult = sourcePresentation.compareWith(targetPresentation, comparisonSettings);
        System.out.println("textFiles was compared.");

        // Saving result of comparison to new textFile
        IComparisonTextFile result = compareResult.getTextFile();
        OutputStream resultStream = new FileOutputStream(resultPath);
        result.save(resultStream);
        resultStream.close();
        System.out.println("Result of comparison was saved to textFile with the folloving source path"
                + resultPath + ".");
        System.out.println("===============================================");
        System.out.println("");
    }

}

File: src/main/resources/text/settings/ComparisonWithDifferentSettings.cs
using System;
using System.IO;
using System.Reflection;
using GroupDocs.Comparison.Common.ComparisonSettings;
using GroupDocs.Comparison.Text;
using GroupDocs.Comparison.Text.Contracts;

namespace Groupdocs.Comparison.Samples.Text.Settings
{
	internal class ComparisonWithDifferentSettings
	{
		public static void CompareTextFilesWithGenerationSummaryPage()
		{
			string resultPath = @"./../../Settings/testresult/WithGenerationSummaryPage/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.GenerateSummaryPage = true;
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithOutGenerationSummaryPage()
		{
			string resultPath = @"./../../Settings/testresult/WithOutGenerationSummaryPage/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.GenerateSummaryPage = false;
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithShowDeletedContent()
		{
			string resultPath = @"./../../Settings/testresult/WithShowDeletedContent/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.ShowDeletedContent = true;
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithOutShowDeletedContent()
		{
			string resultPath = @"./../../Settings/testresult/WithOutShowDeletedContent/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.ShowDeletedContent = false;
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithComparisonDepthSetWords()
		{
			string resultPath = @"./../../Settings/testresult/WithComparisonDepthSetWords/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.ComparisonDepth = ComparisonDepth.Words;
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithComparisonDepthSetChars()
		{
			string resultPath = @"./../../Settings/testresult/WithComparisonDepthSetChars/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.ComparisonDepth = ComparisonDepth.Chars;
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithWordsSepCharsSetSpace()
		{
			string resultPath = @"./../../Settings/testresult/WithWordsSepCharsSetSpace/result.txt";
			// Creating settings for comparison of textFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.WordsSeparatorChars = new char[] {' '};
			Compare(resultPath, comparisonSettings);
		}

		public static void CompareTextFilesWithSettingStylesOnDelInsComponents()
		{
			string resultPath = @"./../../Settings/testresult/WithSettingStylesOnDelInsComponents/result.txt";
			// Creating settings for comparison of TextFiles
			TextComparisonSettings comparisonSettings = new TextComparisonSettings();
			comparisonSettings.DeletedItemsStyle.BeginSeparatorString = "<begin deleted text>";
			comparisonSettings.DeletedItemsStyle.EndSeparatorString = "<end deleted text>";
			comparisonSettings.InsertedItemsStyle.BeginSeparatorString = "<begin inserted text>";
			comparisonSettings.InsertedItemsStyle.EndSeparatorString = "<end inserted text>";
			Compare(resultPath, comparisonSettings);
		}

		private static void Compare(string resultPath, TextComparisonSettings comparisonSettings)
		{
			string sourcePath = @"Groupdocs.Comparison.Samples.Text.Settings.data.source.txt";
			string targetPath = @"Groupdocs.Comparison.Samples.Text.Settings.data.target.txt";
			// Create to streams of textFiles
			Assembly assembly = Assembly.GetExecutingAssembly();
			Stream sourceStream = assembly.GetManifestResourceStream(sourcePath);
			Stream targetStream = assembly.GetManifestResourceStream(targetPath);
			// Opening two textFiles
			IComparisonTextFile sourcePresentation = new ComparisonTextFile(sourceStream);
			Console.WriteLine("TextFile with source path: " + sourcePath + " was loaded.");
			IComparisonTextFile targetPresentation = new ComparisonTextFile(targetStream);
			Console.WriteLine("TextFile with source path: " + targetPath + " was loaded.");

			// Comparing textFiles
			ITextCompareResult compareResult = sourcePresentation.CompareWith(targetPresentation, comparisonSettings);
			Console.WriteLine("textFiles was compared.");

			// Saving result of comparison to new textFile
			IComparisonTextFile result = compareResult.GetTextFile();
			Stream resultStream = new FileStream(resultPath, FileMode.Create);
			result.Save(resultStream);
			resultStream.Close();
			Console.WriteLine("Result of comparison was saved to textFile with the folloving source path" +
			                  resultPath + ".");
			Console.WriteLine("===============================================");
			Console.WriteLine("");
		}
	}
}